blackduck/Client.py

```diff
@@ -102,6 +103,94 @@ def __init__(
         self.base_url = base_url
         self.session = session or HubSession(base_url, timeout, retries, verify)
         self.session.auth = (auth or BearerAuth)(session=self.session, token=token, username=username, password=password)
+        self.root_resources_dict = None
+
+    def list_resources(self, parent=None):
+        """List resources that can be fetched
```

Comment:
Returning a dict instead of a list allows caching (by get_resource) and it is easier for the developer to navigate the API if they see the URLs as well.